exonum-java-binding-core/pom.xml

```diff
@@ -125,6 +128,8 @@
                 <argument>${rust.itSubCrate}</argument>
                 <argument>--color</argument>
                 <argument>always</argument>
+                <argument>--target-dir</argument>
```

Comment:
These unit tests are run with the same features that the native library is built. As I understand, specifying `--target-dir` here will incur a full (?) rebuild even if we move "test-rust-lib-no-resource-manager" to ci-only profile. Shan't we specify `--target-dir` in "test-rust-lib-no-resource-manager" only?

Reply:
If we specify `--target-dir` for `test-rust-lib-no-resource-manager` only it will build but won't override the `libjava_binding` shared library artifact built during the `build-rust-lib` phase and will crash with a dynamic linking error while executing the tests on the `test-rust-lib` phase. And yes - it will cause the full rebuild even if we move `test-rust-lib-no-resource-manager` to ci-only profile.

Reply:
I am not sure I understand — you say that we specify --target-dir for test-rust-lib-no-resource-manager to isolate it from the rest of the builds, but isn't that enough? How a test _with_ a resource manager affects a test _without_ that uses _another_ target directory?

Reply:
Tests _with_ and _without_ resource manager are sharing the same folder (with changes from this PR) specified with `--target-dir`. The main problem with the `test` phase when just upgrading to 1.28.0 step by step: (1) at the `build` step we build the _libjava_binding.dylib_ library, then the `test` phase comes and (2) tests _without_ resource-manager get built and executed against the library built at (1). The same is for (3) tests _with_ resource-manager that also require rebuilt (with cargo test) version of _libjava_binding.dylib_. My current solution is that I specify the same directory for _both_ of tests _with_ and _without_ resource-manager and they are executed one after another and share the resulting binaries from `--target-dir`. If I don't specify the `--target-dir` for tests _with_ resource-manager it will fail anyway because it will attempt to rebuild the _libjava_binding.dylib_ and _wont_ be able to replace the previous version from (1) and will fail with a dynamic linking error.

Reply:
I see, thanks. This part seems strange:

>  If I don't specify the --target-dir for tests with resource-manager it will fail anyway because it will attempt to rebuild the libjava_binding.dylib and wont be able to replace the previous version from (1) and will fail with a dynamic linking error.

Why does it have to replace anything from (1) if it is built with the same flags (= features)? 

The problem with such solution (always separating targets for build and test) is that even if we don't run a build _without_ rm (2) locally — a build _with_ rm (3) will still increase the compilation time significantly.